includes/core/server/request/class-create-intention.php

```diff
@@ -47,7 +47,7 @@ public function get_method(): string {
 	 * @throws Invalid_Request_Parameter_Exception
 	 */
 	public function set_payment_method( string $payment_method_id ) {
-		$this->validate_stripe_id( $payment_method_id, [ 'pm', 'src' ] );
+		$this->validate_stripe_id( $payment_method_id, [ 'pm', 'src', 'card' ] );
```

Comment:
Not urgent but I think we might want to add more tests to cover for this change: 

https://github.com/Automattic/woocommerce-payments/blob/96a1efd7b9e94491a34c0f929661d9c8f636e353/tests/unit/core/server/request/test-class-core-create-and-confirm-intention-request.php#L79-L83

Also, this method is being used in the child class `Create_And_Confirm_Intention` so we might lift it up to this class?